handout/testing/services/mockbackend.md

```diff
@@ -0,0 +1,103 @@
+# Testing HTTP Requests Using MockBackend
+
+Since we want to isolate our service testing from outside network requests, we have to create mock requests where we can manually set up properties like expected responses, errors, headers, etc. Fortunately Angular 2 makes this an easy process by providing a way to make fake HTTP calls through the `MockBackend` class. This class can be injected wherever a service is expecting the HTTP module and used by the service as though it were the real HTTP module - except now the results/data are set up by us, without any network request taking place.
+
+We want to use `MockBackend` to inject a mocked version of HTTP, so that any call to `http.get` will allow us to feed in whatever data we want - not the real data from wikipedia.org. Lets take a look at our unit test:
+
+*wikisearch.spec.ts*
+
+```js
+import {
+  fakeAsync,
+  inject,
+  TestBed
+} from '@angular/core/testing';
+import {
+  HttpModule,
+  XHRBackend,
+  ResponseOptions,
+  Response
+} from '@angular/http';
+import {
+  MockBackend,
+  MockConnection
+} from '@angular/http/testing/mock_backend';
+
+import {SearchWiki} from './wikisearch.service';
+
+const mockResponse = {
+  "batchcomplete": "",
+  "continue": {
+    "sroffset": 10,
+    "continue": "-||"
+  },
+  "query": {
+    "searchinfo": {
+      "totalhits": 36853
+    },
+    "search": [{
+      "ns": 0,
+      "title": "Stuff",
+      "snippet": "<span></span>",
+      "size": 1906,
+      "wordcount": 204,
+      "timestamp": "2016-06-10T17:25:36Z"
+    }]
+  }
+};
+
+describe('Wikipedia search service', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpModule],
+      providers: [
+        {
+          provide: XHRBackend,
+          useClass: MockBackend
+        },
+        SearchWiki
+      ]
+    });
+  });
+
+  it('should get search results', fakeAsync(
+    inject([
+      XHRBackend,
+      SearchWiki
+    ], (mockBackend, searchWiki) => {
```

Comment:
I think so, otherwise we will not get code completion from the IDE.

```js
inject(
  [XHRBackend, SearchWiki], 
  (mockBackend: MockBackend, searchWiki: SearchWiki) => {
    ...
  }
)
```